astropy/units/decorators.py

```diff
@@ -91,7 +91,7 @@ def _validate_arg_value(
             "be in units convertible to"
         )
         if len(targets) > 1:
-            targ_names = ", ".join([f"'{str(targ)}'" for targ in targets])
+            targ_names = ", ".join([f"'{targ}'" for targ in targets])
             raise UnitsError(f"{error_msg} one of: {targ_names}.")
         else:
             raise UnitsError(f"{error_msg} '{str(targets[0])}'.")
```

Comment:
I wonder why the `str()` inside `{}` is not picked up here... But if everything else is automatic, fine with leaving it.

Reply:
It tried turning it into
``targ_names = ", ".join([f"'{targ!s}'" for targ in targets])``,
but the ``!s`` is the default and necessary to specify.

Reply:
I meant why not also do `raise UnitsError(f"{error_msg} '{targets[0]}'.")`

Reply:
Oh, yes, the other error could also be simplified. Strange that ruff didn't pick up on that!